qiskit/quantum_info/operators/operator.py

```diff
@@ -391,6 +392,8 @@ def from_circuit(
             final_layout (Layout): If specified this kwarg can be used to represent the
                 output permutation caused by swap insertions during the routing stage
                 of the transpiler.
+            original_qubit_indices (dict): The mapping from qubits to positional indices
+                for the ``layout`` argument.
```

Comment:
This argument allows creating a callback function that checks the equivalence of the current operator and the original operator after each transpiler pass, something along the following:

```
    def callback_func(pass_, dag, time, property_set, count):
        circuit = dag_to_circuit(dag)
        current_op = Operator.from_circuit(
            circuit,
            layout=property_set['layout'],
            final_layout=property_set['final_layout'],
            original_qubit_indices=property_set['original_qubit_indices']
        )

        current_op_new = Operator._from_circuit_new(
            circuit,
            layout=property_set['layout'],
            original_qubit_indices=property_set['original_qubit_indices']
        )

        original_op = op if circuit.num_qubits == original_num_qubits else extended_op

        print(f"After-pass: equiv =  {original_op.equiv(current_op)}")
        print(f"After-pass-new: equiv =  {original_op.equiv(current_op_new)}")
```
I have found this very useful to catch problems, however this is still not fully correct: the `SetLayout` pass sets the pass manager's "layout" attribute, but it's only the later pass `ApplyLayout` that effectively applies it to the circuit, so in-between these two passes the equivalence is not preserved. Though a simple way would be to have `ApplyLayout` set an extra attribute `layout_is_applied` and use `layout` when checking equivalence only when `layout_is_applied` is set.